### PR TITLE
[18.01] Add support for the ideaspre datatype

### PR DESCRIPTION
**Pull request by gregvonkuster**
@martenson this looks messy to me, but I'm not a cherry-pick expert, so I hope this is ok.  Please let me know.  Thanks a bunch for the help on this!

### CODE REVIEW

**Comment by martenson** [comment]
Yeah, this does not look right. And it is targeting dev again.

**Comment by mvdbeek** [comment]
Targeting dev was the problem, the rest is OK

**Comment by gregvonkuster** [comment]
Ah, thanks to both of you!

File: lib/galaxy/datatypes/genetics.py [hunk 1]
```diff
@@ -518,6 +518,73 @@ class Snptest(Rgenetics):
     file_ext = "snptest"
 
 
+class IdeasPre(Rgenetics):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/genetics.py]: You can probably use `Html` here instead of `Rgenetics`

Comment (martenson) [reply lib/galaxy/datatypes/genetics.py]: @nsoranzo https://github.com/galaxyproject/planemo/issues/760

Comment (nsoranzo) [reply lib/galaxy/datatypes/genetics.py]: Why would that be the case?

Comment (gregvonkuster) [reply lib/galaxy/datatypes/genetics.py]: @nsoranzo The main reason is that the Galaxy upload tool will not allow uploading the Html datatype - https://github.com/galaxyproject/galaxy/blob/dev/tools/data_source/upload.py#L277.  But tool functional tests require the use of the upload tool.  I suppose this class could subclass the Html datatype like the Rgenetics datatype does though.  Is this needed?

Comment (nsoranzo) [reply lib/galaxy/datatypes/genetics.py]: @gregvonkuster That was my point, subclassing Rgenetics and overload all its method is equivalent to subclass Html. But I see you changed it already, thanks!

**Comment by gregvonkuster** [comment]
Ping @davebx and @martenson is this ready for merging?

**Comment by gregvonkuster** [comment]
Ping @davebx and @martenson can this be merged?

**Comment by davebx** [comment]
@gregvonkuster yes it can.

**Comment by gregvonkuster** [comment]
Thanks a bunch @davebx ;)

**Pull request merged**